rapids_build_backend/impls.py

```diff
@@ -271,27 +161,45 @@ def _edit_pyproject(config):
     being built. This is useful for projects that want to build wheels
     with a different name than the package name.
     """
-    pyproject = _get_pyproject()
-    project_data = pyproject["project"]
-    project_data["name"] += _get_cuda_suffix(config.require_cuda)
-
-    dependencies = pyproject["project"].get("dependencies")
-    if dependencies is not None:
-        project_data["dependencies"] = _process_dependencies(
-            config, project_data["dependencies"]
-        )
-
-    optional_dependencies = pyproject["project"].get("optional-dependencies")
-    if optional_dependencies is not None:
-        project_data["optional-dependencies"] = {
-            extra: _process_dependencies(config, deps)
-            for extra, deps in optional_dependencies.items()
-        }
-
     pyproject_file = "pyproject.toml"
     bkp_pyproject_file = ".pyproject.toml.rapids-build-backend.bak"
+
+    cuda_version = _get_cuda_version(config.require_cuda)
+
+    parsed_config = rapids_dependency_file_generator.load_config_from_file(
+        config.dependencies_file
+    )
+
     try:
-        shutil.move(pyproject_file, bkp_pyproject_file)
+        shutil.copyfile(pyproject_file, bkp_pyproject_file)
+        for file_key, file_config in parsed_config.files.items():
+            if (
+                rapids_dependency_file_generator.Output.PYPROJECT
+                not in file_config.output
+            ):
+                continue
+            pyproject_dir = os.path.join(
+                os.path.dirname(config.dependencies_file),
+                file_config.pyproject_dir,
+            )
+            if not os.path.exists(pyproject_dir):
+                continue
+            if not os.path.samefile(pyproject_dir, "."):
+                continue
+            matrix = _parse_matrix(config.matrix) or dict(file_config.matrix)
+            if cuda_version is not None:
+                matrix["cuda"] = [f"{cuda_version[0]}.{cuda_version[1]}"]
+            rapids_dependency_file_generator.make_dependency_files(
+                parsed_config,
+                [file_key],
+                {rapids_dependency_file_generator.Output.PYPROJECT},
+                matrix,
+                [],
+                False,
+            )
```

Comment:
I haven't reviewed this thoroughly, just leaving a drive-by comment... I strongly recommend the following changes:

* for any `rapids_dependency_file_generator` function in this project that takes more than 1 argument, call it exclusively with keyword arguments
* change the function signatures throughout `rapids_dependency_file_generator` to enforce that, like this:

```python
def make_dependency_files(*, arg1, arg2, arg3)
```

That'd eliminate the risk of any "oops I passed this in the wrong order" types of bugs. It'd also make calls like this one easier to read.

Here's an example of that pattern: https://github.com/rapidsai/dependency-file-generator/pull/65